opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -300,32 +300,31 @@ updateGuideRatesForInjectionGroups(const Group& group,
         if(!group.hasInjectionControl(phase))
             continue;
 
-        Scalar guideRateValue = 0.0;
+        std::optional<Scalar> guideRateValue;
         const auto& controls = group.injectionControls(phase, summaryState);
         switch (controls.guide_rate_def){
         case Group::GuideRateInjTarget::RATE:
             break;
         case Group::GuideRateInjTarget::VOID:
         {
-            guideRateValue = group_state.injection_vrep_rate(group.name());
+            guideRateValue = std::max(Scalar(0.0), group_state.injection_vrep_rate(group.name()));
```

Comment:
Why is the max() needed here? Also added on line 324.

Reply:
we assume that all guiderates are positive when we compute the fractions. In the extreme case of all wells have zero target I though that we may end of with negative rates due to lack of numerical precision. The max is just a precaution. We can probably remove it and nobody will notice in a 100 year.